src/entities/road.lua

```diff
@@ -25,19 +52,84 @@ function road:draw()
                        segment.body:getAngle())
   end
   love.graphics.polygon("fill",
-    self.frontier.body:getWorldPoints(self.frontier.shape:getPoints())
+    self.frontier.main.body:getWorldPoints(self.frontier.main.shape:getPoints())
+  )
+  love.graphics.setColor(0, 255, 0, 255)
+  love.graphics.polygon("fill",
+    self.frontier.right.body:getWorldPoints(self.frontier.right.shape:getPoints())
   )
+  love.graphics.setColor(255, 0, 0, 255)
+  love.graphics.polygon("fill",
+    self.frontier.left.body:getWorldPoints(self.frontier.left.shape:getPoints())
+  )
+
 end
 
-function road:update(x, y, angle)
-  self.frontier.body:setAngle(angle)
-  fy = (self.roadjog) * math.sin(self.frontier.body:getAngle())
-  fx = (self.roadjog) * math.cos(self.frontier.body:getAngle())
-  self.frontier.body:setX(x + fx)
-  self.frontier.body:setY(y + fy)
+function road:update(angle, roadShift)
   self:addsegment()
+
+  mainX = self.frontier.main.body:getX()
+  mainY = self.frontier.main.body:getY()
+  
+  lastAngle = self.frontier.main.body:getAngle()
+  deltaAngle = lastAngle - angle
+  if math.abs(deltaAngle) > math.pi then
+    if deltaAngle > 0 then
+      deltaAngle = deltaAngle - 2 * math.pi
+    else
+      deltaAngle = deltaAngle + 2 * math.pi
+    end
+  end
+  
+  if math.abs(deltaAngle) > math.pi/2 then
+    --car crashes? this should never happen unless people get off the track
+  end
+  
+  if math.abs(deltaAngle) > self.rotationLimit then
```

Comment:
Same thing here; I think my first comment correctly guessed what this was used for, but it is not necessarily apparently after the manipulation done above to `deltaAngle`.

Reply:
Commented. Let me know if this documentation is clear enough